docs/design/rosbag2_record_replay_action.md

```diff
@@ -0,0 +1,136 @@
+# Rosbag2 record and replay action
+
+## Context
+
+This design describes how to implement this feature based on current architecture and how to extend current command parameter.
+
+## How to record action message
+
+The implementation of the action uses three services and two topics. For the services, when service introspection is enabled, rosbag2 can record service data through the service event topic (refer to [rosbag2_record_replay_service.md](./rosbag2_record_replay_service.md)). So all data related to the action can be recorded.
+
+
+## Which topics should be focused on for an action
+
+- goal service
+
+  [NameSpace/]ActionName/_action/send_goal  (Service Event Topic: [NameSpace/]ActionName/_action/send_goal/_service_event)
+
+- result service
+
+  [NameSpace/]ActionName/_action/get_result  (Service Event Topic: [NameSpace/]ActionName/_action/get_result/_service_event)
+
+- cancel service
+
+  [NameSpace/]ActionName/_action/cancel_goal  (Service Event Topic: [NameSpace/]ActionName/_action/cancel_goal/_service_event)
+
+- status topic
+
+  [NameSpace/]ActionName/_action/status
+
+- feedback topic
+
+  [NameSpace/]ActionName/_action/feedback
```

Comment:
Please check how this section is rendered on the web. At least the GitHub review panel renders it a bit differently than expected.

Reply:
I didn't notice any differences. The [ ] indicates optional.  Could you please explain in detail?

![Screenshot from 2025-03-12 17-43-48](https://github.com/user-attachments/assets/41051f53-0599-4c53-948a-3c087a8419fb)

Reply:
![image](https://github.com/user-attachments/assets/e52d8d8b-e8af-4d14-b310-7fa7b17e0048)

Reply:
Thanks. I will update the contents.